provider/aws/aws_discover.go

```diff
@@ -16,6 +23,34 @@ import (
 
 type Provider struct{}
 
+const ECSMetadataURIEnvVar = "ECS_CONTAINER_METADATA_URI_V4"
+
+type ECSTaskMeta struct {
+	Cluster    string                 `json:"Cluster"`
+	TaskARN    string                 `json:"TaskARN"`
+	Family     string                 `json:"Family"`
+	Containers []ECSTaskMetaContainer `json:"Containers"`
+}
+
+type ECSTaskMetaContainer struct {
+	Name          string               `json:"Name"`
+	Health        ECSTaskMetaHealth    `json:"Health"`
+	DesiredStatus string               `json:"DesiredStatus"`
+	KnownStatus   string               `json:"KnownStatus"`
+	Networks      []ECSTaskMetaNetwork `json:"Networks"`
+}
+
+type ECSTaskMetaHealth struct {
+	Status      string `json:"status"`
+	StatusSince string `json:"statusSince"`
+	ExitCode    int    `json:"exitCode"`
+}
+
+type ECSTaskMetaNetwork struct {
+	IPv4Addresses  []string `json:"IPv4Addresses"`
+	PrivateDNSName string   `json:"PrivateDNSName"`
+}
```

Comment:
nit: Since I think this only uses the TaskARN, we could remove the other unused fields. Basically, just this:

```suggestion
type ECSTaskMeta struct {
	TaskARN string `json:"TaskARN"`
}
```